src/sage/matrix/matrix_polynomial_dense.pyx

```diff
@@ -1662,20 +1662,30 @@ cdef class Matrix_polynomial_dense(Matrix_generic_dense):
         if self.ncols() == 0 or self.nrows() == 0:
             return self._is_empty_popov(row_wise)
         leading_positions = self.leading_positions(shifts, row_wise)
-        # here, it will be convenient to have leading position
-        # larger than ncols for zero/empty rows
-        leading_positions = [pos if pos>=0 else self.ncols() + 1 for pos in leading_positions]
+        # here, because of the below sorting and of the convention that zero
+        # rows (resp. columns) are at the bottom (resp. right) of the matrix in
+        # the row-wise case (resp. column-wise case), it will be convenient to
+        # have leading position ncols (resp. nrows) for these zero vectors
+        pos_zero_vec = self.ncols() if row_wise else self.nrows()
+        leading_positions = [pos if pos>=0 else pos_zero_vec + 1 \
+                                         for pos in leading_positions]
         # leading positions should not have duplicates, which is equivalent to:
         # once sorted, it doesn't contain a pair of equal successive entries
+        # (we do not sort if we want to test
```

Comment:
I guess this line is a leftover from a previous version of the comment starting at line 1678 and should be removed? Lgtm otherwise.

Reply:
Yes, absolutely, thanks for spotting this.